cmd/deploy.go

```diff
@@ -13,24 +14,58 @@ type DeployCmd struct {
 	ui              boshui.UI
 	deployment      boshdir.Deployment
 	releaseUploader ReleaseUploader
+	director        boshdir.Director
 }
 
 type ReleaseUploader interface {
 	UploadReleases([]byte) ([]byte, error)
 	UploadReleasesWithFix([]byte) ([]byte, error)
 }
 
+type Conf struct {
+	Flags              []string `yaml:"flags"`
+	IncludeDeployments []string `yaml:"include"`
+	ExcludeDeployments []string `yaml:"exclude"`
+}
+
 func NewDeployCmd(
 	ui boshui.UI,
 	deployment boshdir.Deployment,
 	releaseUploader ReleaseUploader,
+	director boshdir.Director,
 ) DeployCmd {
-	return DeployCmd{ui, deployment, releaseUploader}
+	return DeployCmd{ui, deployment, releaseUploader, director}
 }
 
 func (c DeployCmd) Run(opts DeployOpts) error {
 	tpl := boshtpl.NewTemplate(opts.Args.Manifest.Bytes)
 
+	configs, _ := c.director.ListConfigs(1, boshdir.ConfigsFilter{Type: "deploy"})
+
+	for _, config := range configs {
+		var conf Conf
+
+		yaml.Unmarshal([]byte(config.Content), &conf)
+
+		deploymentIncluded := containsDeployment(conf.IncludeDeployments, c.deployment.Name())
+		deploymentExcluded := containsDeployment(conf.ExcludeDeployments, c.deployment.Name())
+
+		if conf.ExcludeDeployments != nil &&
+			conf.IncludeDeployments != nil {
+			c.ui.PrintLinef("Ignoring deployment flags from config of type '%s' (name: '%s'). Please use only 'include'- OR 'exclude'-property in the config.", config.Type, config.Name)
+		} else {
+			if (conf.IncludeDeployments == nil && conf.ExcludeDeployments == nil) ||
+				deploymentIncluded ||
+				(!deploymentExcluded && conf.ExcludeDeployments != nil) {
+				c.ui.PrintLinef("!!!!!!!")
+				c.ui.PrintLinef("Using deployment flags from config of type '%s' (name: '%s')", config.Type, config.Name)
+				c.ui.PrintLinef("!!!!!!!")
```

Comment:
Message should be improved to be similar to other Bosh CLI outputs and still raise the relevant attention from human operators